### ISSUE TITLE
fix: shift-tab does not work in FluentPopover, and tab/shift-tab on the popover anchor while popover is open does not close it

### ISSUE THREAD

**Issue by adamint** [microsoft/fast-blazor#4127]
# 🐛 Bug Report

When navigating through a popover with the keyboard, shift + tab should go to the previous element. Instead, it moves to the next element in the popover.

Also, tabbing/shift-tab with the popover open while focus is on the popover **anchor** does not close the popover, as it should.

Repro by focusing `Open Callout 1` and pressing tab/shift-tab -> https://www.fluentui-blazor.net/Popover

(a mix of shift+tab and tab were used in this video, the fact that you cannot tell is the problem!)

https://github.com/user-attachments/assets/666cb49f-cc79-4d17-af66-b4eae401f95e

## 💁 Possible Solution

The code in https://github.com/microsoft/fluentui-blazor/pull/4093 could be tweaked for the popover component? It handles both of the cases at issue here.

## 🔦 Context

https://github.com/dotnet/aspire/issues/10129

**Comment by adamint** [comment]
@vnbaaij when #4093 is merged, I would be happy to contribute a fix myself.

**Comment by vnbaaij** [comment]
> @vnbaaij when #4093 is merged, I would be happy to contribute a fix myself.

I've just approved that PR. Will merge once checks are done

**Comment by vnbaaij** [comment]
Hi @adamint. We just finished and merged #4093. We are awaiting your PR for this one😁

**Comment by vnbaaij** [comment]
Hi @adamint,

Are you still planning to do a PR for this?
